### PR TITLE
Remove mongo collection instance package

### PR DESCRIPTION
**Pull request by lekev**
Remove no needed package mongo-collection-instance . It also fix the issue https://github.com/Urigo/angular-meteor/issues/644

### CODE REVIEW

**Comment by Urigo** [comment]
@lekev you are right. 
The question is, does that change makes the `angular` package depends on a new version of Meteor?
Which means we should change that - https://github.com/Urigo/angular-meteor/blob/master/packages/angular/package.js#L10
and if so, to what version and do we really want that?
maybe we do, I'm not sure.  what do you think?

**Comment by lekev** [comment]
the changes is not dependant of the new meteor version,  Since the AngularMeteorCollection is deprecated and replace by the helpers function, we will not use anymore the mongo-collection-instance . 
Actually we still use it [here](https://github.com/Urigo/angular-meteor/blob/496b1d00cbf2db9fa2df71b30d3eeae1f4c5489d/packages/angular-meteor-data/modules/angular-meteor-collection.js#L41) . So maybe should I find a way to replace $meteorUtils.getCollectionByName to not use it, until you release the version 1.4.0

**Comment by lekev** [comment]
I am already use my fork with meteor 1.2.1 , and it works fine

File: packages/angular-meteor-data/modules/angular-meteor-collection.js [hunk 1]
```diff
@@ -34,12 +34,7 @@ angularMeteorCollection.factory('AngularMeteorCollection', [
       // (autorun for cursor.fetch).
       data._hDataAutorun = null;
 
-      if (angular.isDefined(collection)) {
-        data.$$collection = collection;
-      } else {
-        var cursor = curDefFunc();
-        data.$$collection = $meteorUtils.getCollectionByName(cursor.collection.name);
-      }
+      data.$$collection = collection;
```
Comment (Urigo) [inline packages/angular-meteor-data/modules/angular-meteor-collection.js]: @lekev The thing is that we need this `if` statement so you could define both a cursor or a string.
How do you handle defining a Cursor from a string here?

**Comment by Urigo** [comment]
@lekev In 1.3 I still want and need to support AngularMeteorCollection.
For that I have to be able to get the instance of the Collection by it's string name.
your code would make AngularMeteorCollection not usable.  have you tested the 1.2 API?

**Pull request closed without merging**